### PR TITLE
[improve][ci] Migrate from Gradle Enterprise to Develocity

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

We are using Gradle's Develocity product for build analytics. ASF's server is at  https://ge.apache.org/ .
Gradle Develocity used to be called Gradle Enterprise. This PR contains the [migration from legacy Gradle Enterprise Maven extension](https://docs.gradle.com/develocity/maven-extension/legacy/#develocity_migration) to [Develocity Maven Extension](https://docs.gradle.com/develocity/maven-extension/current/).

### Modifications

- Migrate files from Gradle Enterprise Maven extension to Develocity Maven Extension format.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -95,5 +95,5 @@ pulsar-client/src/test/java/org/apache/pulsar/client/avro/generated
 
 test-reports/
 
-# Gradle Enterprise
-.mvn/.gradle-enterprise/
```
Comment (heesung-sn) [inline .gitignore:99]: should we include both instead of replacing it?

File: .mvn/develocity.xml [hunk 2]
```diff
@@ -19,25 +19,22 @@
     under the License.
 
 -->
-<gradleEnterprise
-        xmlns="https://www.gradle.com/gradle-enterprise-maven" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-        xsi:schemaLocation="https://www.gradle.com/gradle-enterprise-maven https://www.gradle.com/schema/gradle-enterprise-maven.xsd">
-  <!-- Enable Gradle Enterprise extension when GRADLE_ENTERPRISE_ACCESS_KEY is set and the build isn't
+<develocity xmlns="https://www.gradle.com/develocity-maven" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="https://www.gradle.com/develocity-maven https://www.gradle.com/schema/develocity-maven.xsd">
+<!-- Enable Gradle Enterprise extension when GRADLE_ENTERPRISE_ACCESS_KEY is set and the build isn't
```
Comment (heesung-sn) [inline .mvn/develocity.xml:24]: should we update this comment (e.g. Enable Develocity ...)?

File: .mvn/develocity.xml [hunk 3]
```diff
@@ -19,25 +19,22 @@
...
   </server>
   <buildScan>
     <capture>
-      <goalInputFiles>true</goalInputFiles>
       <buildLogging>true</buildLogging>
       <testLogging>true</testLogging>
     </capture>
     <backgroundBuildScanUpload>#{isFalse(env['GITHUB_ACTIONS'])}</backgroundBuildScanUpload>
-    <publish>ALWAYS</publish>
-    <publishIfAuthenticated>true</publishIfAuthenticated>
```
Comment (heesung-sn) [inline .mvn/develocity.xml:40]: nit: I am wondering why we need to update these configs? Could you explain a little here?

**Review by heesung-sn — APPROVED** [review]

File: .mvn/develocity.xml (hunk 3 above)
Comment (lhotari) [reply .mvn/develocity.xml:40]: [followed the migration guide](https://docs.gradle.com/develocity/maven-extension/legacy/#develocity_migration) and checked in IntelliJ that the new XML was following the new XML schema. These elements have been removed / replaced and that's why I removed them.

File: .mvn/develocity.xml (hunk 2 above)
Comment (lhotari) [reply .mvn/develocity.xml:24]: I'll do that later, once ASF Infra has switched to use DEVELOCITY_ACCESS_KEY.

File: .gitignore (hunk 1 above)
Comment (lhotari) [reply .gitignore:99]: valid point

File: .mvn/develocity.xml (hunk 2 above)
Comment (lhotari) [reply .mvn/develocity.xml]: Updated

File: .gitignore (hunk 1 above)
Comment (lhotari) [reply .gitignore:99]: Done

**Pull request merged**